### PR TITLE
Add method to send post with multiple attached photos

### PR DESCRIPTION
**Pull request by ryoryo25**
## Description
I've added a function `send_images` to send a post with multiple attached images.

## Changes
- Add function `send_images` to class Client & AsyncClient
- Rewrite function `send_image` using `send_images` (to keep backward compatibility)
- Add an example of usage of `send_images`

### CODE REVIEW

**Comment by MarshalX** [comment]
Hi! Thank you for contributing! I saw that many people looking for this method. Backward compatibility is awesome. A few notes before deep diving into code:

- we write sync Client and then auto-generate async version. Looks like the generator failed. Probably you wrote the async version by hand. Could you pls fix it?
- speaking of blobls upload in the async version could we try to upload blobs simultaneously? For example, using asyncio.gather. I mean and don't see the reason to wait to upload the previous image to upload a new one 
- check on 4 images and no more should be covered already by the model. I guess we could delete this if from here

**Comment by ryoryo25** [comment]
Thank you for reviewing my code!
I revised the following parts you mentioned.
- Revert the async version of Client
- Delete validation for the number of images
- Modify the code generator of AsyncClient to generate code for uploading blobs simultaneously by using asyncio.gather
  ```python
  uploads = [self.upload_blob(i) for i in images]
  ```
  converts to
  ```python
  uploads = asyncio.gather(*[self.upload_blob(i) for i in images])
  ```

File: packages/atproto_client/client/client.py [hunk 1]
```diff
@@ -173,6 +174,58 @@ def delete_post(self, post_uri: str) -> bool:
...
+            :class:`atproto.exceptions.AtProtocolError`: Base exception.
+        """
+        if image_alts is None:
+            image_alts = [''] * len(images)
+
+        # validation
+        if len(images) != len(image_alts):
+            raise AtProtocolError('The lengths of images & image_alts must be equal.')
+
+        uploads = [self.upload_blob(i) for i in images]
```
[the remark this answers is not in the document]
Comment (ryoryo25) [reply packages/atproto_client/client/client.py]: fixed with 3cc8bc1407534e6226e52aed3cb1a8906cde8a28

File: packages/atproto_client/client/client.py [hunk 2]
```diff
@@ -173,6 +174,58 @@ def delete_post(self, post_uri: str) -> bool:
...
+        embed_images = [
+            models.AppBskyEmbedImages.Image(alt=a, image=u.blob)
+            for a, u in zip(image_alts, uploads)
```
[the remark this answers is not in the document]
Comment (ryoryo25) [reply packages/atproto_client/client/client.py]: fixed with 3cc8bc1407534e6226e52aed3cb1a8906cde8a28

File: packages/atproto_client/client/client.py [hunk 3]
```diff
@@ -173,6 +174,58 @@ def delete_post(self, post_uri: str) -> bool:
...
+
+        Raises:
+            :class:`atproto.exceptions.AtProtocolError`: Base exception.
+        """
+        if image_alts is None:
+            image_alts = [''] * len(images)
+
+        # validation
+        if len(images) != len(image_alts):
+            raise AtProtocolError('The lengths of images & image_alts must be equal.')
```
[the remark this answers is not in the document]
Comment (ryoryo25) [reply packages/atproto_client/client/client.py]: fixed with 3cc8bc1407534e6226e52aed3cb1a8906cde8a28

File: packages/atproto_codegen/clients/generate_async_client.py [hunk 4]
```diff
@@ -40,6 +41,12 @@ def gen_client(input_filename: str, output_filename: str) -> None:
         code = code.replace(f'self.{method}(', f'await self.{method}(')
         code = code.replace(f'super().{method}(', f'await super().{method}(')
 
+    code = code.replace('from asyncio', 'import asyncio\nfrom asyncio')
```
[the remark this answers is not in the document]
Comment (ryoryo25) [reply packages/atproto_codegen/clients/generate_async_client.py]: fixed with 156dde7a04fb911cad7d3d28219af0cbea8b17b9

File: examples/send_images.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    paths = ['cat.jpg', 'dog.jpg', 'bird.jpg']
+    images = []
+    for path in paths:
+        with open(path, 'rb') as f:
+            images.append(f.read())
+
+    client.send_images(
+        text='Post with image from Python',
+        images=images,
+        image_alts=['Text version', 'of the image (ALT)', 'This parameter is optional'])
```
[the remark this answers is not in the document]
Comment (ryoryo25) [reply examples/send_images.py]: fixed with 68c1a73423320385964e7178a01a9e7b07bc45cd

**Comment by ryoryo25** [comment]
I've fixed the parts you mentioned. I think CI would be passed.

**Review by MarshalX — APPROVED** [review]
great job!

**Pull request merged**